client/finality-grandpa/src/finality_proof.rs

```diff
@@ -145,6 +147,55 @@ impl<Block: BlockT> AuthoritySetForFinalityChecker<Block> for Arc<dyn FetchCheck
 	}
 }
 
+
+/// Justification for a finalized block.
+#[derive(Clone)]
+pub struct JustificationNotification<Block: BlockT> {
+	/// Highest finalized block header
+	pub header: Block::Header,
+	/// An encoded justification proving that the given header has been finalized
+	pub justification: Vec<u8>,
+}
+
+type JustificationStream<Block> = TracingUnboundedReceiver<JustificationNotification<Block>>;
+pub type SharedFinalitySubscribers<T> = Arc<RwLock<Vec<JustificationStream<T>>>>;
+
+type FinalityNotifier<T> = TracingUnboundedSender<JustificationNotification<T>>;
+pub type SharedFinalityNotifiers<T> = Arc<RwLock<Vec<FinalityNotifier<T>>>>;
```

Comment:
I'm not a fan of naming here, I think we shouldn't really expose raw list of notifiers - instead this should be encpasulated in some `Subscriptions` object (as below) that is being passed around and best if it's split into receiving and sending halfs, so that the RPC can: `GrandpaJustifications::justification_stream() -> Stream` and grandpa can only do `GrandpaJustificationSubscribers::notify()`.
Internally both halfs will most likely do share that state, but it should be nicely encapsulated and never exposed.